components/brave_rewards/resources/ui/reducers/rewards_reducer.ts

```diff
@@ -47,6 +47,11 @@ const rewardsReducer: Reducer<Rewards.State | undefined> = (state: Rewards.State
       if (key) {
         state[key] = value
         chrome.send('brave_rewards.saveSetting', [key, value.toString()])
+
+        // Disable Ads as well if Rewards has been disabled
+        if (key === 'enabledMain' && !value) {
```

Comment:
I think that we shouldn't disable ads if user didn't disable it, but ads library should check if rewards are enabled or not cc @bridiver @tmancey

Reply:
@NejcZdovc if we want to accomplish this at the native-ads level, let me know and I can remove this change, as the rest of the PR solves a different issue and I don't want it to get held up.

Reply:
I think we should handle this on native level, but would like others to chime in. You can split this PR into two so that first part is not held back as you said

Reply:
@NejcZdovc I've rolled back this change. As for my two cents, I think checking at the native level is important, but I also think it is valuable for the user to see that Ads is visibly disabled on `brave://rewards` too, so if others feel the same way we'd need a change of this sort to land.

Reply:
but if rewards are disabled all boxes are disabled automatically, so user would see disabled ads

Reply:
Indeed, that is correct :)